build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -0,0 +1,280 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fetch from "node-fetch";
+import { promises as fsPromises } from "fs";
+import { Flags } from "@oclif/core";
+import { Logger } from "@fluidframework/build-tools";
+import { BaseCommand } from "../../base";
+import { PackageVersionList } from "../../lib";
+import { isInternalVersionRange } from "@fluid-tools/version-tools";
+
+// Define the interface for build details
+interface IBuildDetails {
+	definition: { name: string };
+	status: string;
+	result: string;
+	sourceBranch: string;
+	finishTime: string;
+	buildNumber: string;
+}
+
+export class UnreleasedReportCommand extends BaseCommand<typeof UnreleasedReportCommand> {
+	static description = `Creates a release report for the most recent build published to an internal ADO feed. It does this by finding the most recent build in ADO produced from a provided branch, and creates a report using that version. The report always uses the "caret" report format.`;
```

Comment:
Make the `static`s here `static readonly`. Craig pointed this out recently and it will improve type safety.